### ISSUE TITLE
Docs: Clarify multiple bboxes behavior in SAM3SemanticPredictor

### ISSUE THREAD

**Issue by Haloukaidi** [ultralytics/ultralytics#24801]
The SAM 3 documentation currently shows:

```python
# Multiple bounding boxes for different concepts
results = predictor(bboxes=[[539, 599, 589, 639], [343, 267, 499, 662]])
```

However, the current `SAM3SemanticPredictor` implementation appears to treat all provided bounding boxes as exemplars for a single visual concept:

```python
nc = 1 if bboxes is not None else len(text) if text is not None else len(self.model.names)
```

The code then creates one geometric prompt and appends all boxes into it, while `text_ids=torch.arange(nc)` contains only one concept id when `bboxes` is provided.

This makes the current example comment misleading. A more accurate comment would be:

```python
# Multiple bounding boxes as exemplars for the same visual concept
results = predictor(bboxes=[[539, 599, 589, 639], [343, 267, 499, 662]])
```

If different concepts are intended, users should either use multiple text prompts or run separate bbox-exemplar queries per concept. A flat `bboxes` list does not currently encode which bbox belongs to which concept.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Haloukaidi, thank you for raising this documentation clarification for Ultralytics 🚀! This is an automated response to help get your issue triaged quickly, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) to help us debug it.

If this is a documentation/reporting issue 📝, thank you for pointing out the potentially misleading example and implementation detail. If possible, please share any additional context such as the exact docs page or section you are referring to, along with the relevant package version, to help us review it faster.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Hello @Haloukaidi 

Thanks for spotting this.

Can you open a PR with your suggested update?

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has now been merged in [PR #24807: Clarify multiple bounding boxes behavior in SAM3 semantic predictor docs](https://github.com/ultralytics/ultralytics/pull/24807).

This change is documentation-only, but it may resolve the confusion reported here by clarifying the intended meaning of multiple bounding boxes in the SAM 3 example:

- The comment in `docs/en/models/sam-3.md` was updated
- The wording was changed from “multiple bounding boxes for different concepts”
- It now correctly explains them as “multiple bounding boxes as exemplars of the same visual concept”
- The example code itself was not changed, only the explanatory text

This should make it clearer that the boxes are meant to represent examples of one shared concept rather than separate concepts, which is likely the source of the issue described here.

Credit for the fix goes to @raimbekovm, with contributions from @Haloukaidi and @onuralpszr.

If you'd like to test the latest changes now, you can install the current main branch with:

```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```

Or, if preferred, you can wait for the next official PyPI release that includes this merged update.

If possible, please check whether this clarification resolves the issue on your side and share the result here. Thanks 🙏 for reporting this, and any additional feedback is welcome if the issue still persists.
